### ISSUE TITLE
apollo-link-state": "0.4.2" and big javascript types/graphql in final bundle.

### ISSUE THREAD

**Issue by frederikhors** [apollographql/apollo-link#796]
**UPDATE:**

The guilty is:

**`"apollo-link-state": "0.4.2"`**

**Is there a fix already?**

------------------------------------------------------------------

I found in my React App these javascript dependencies:

![image](https://user-images.githubusercontent.com/41120635/45749864-561ae500-bc0d-11e8-8053-8b528f687ad9.png)

I think them are from `@types/graphql`.

In my `yarn.lock` file it is just here:

**yarn.lock:**

```
apollo-link@^1.0.0, apollo-link@^1.2.1, apollo-link@^1.2.2:
  version "1.2.2"
  resolved "https://registry.yarnpkg.com/apollo-link/-/apollo-link-1.2.2.tgz#54c84199b18ac1af8d63553a68ca389c05217a03"
  dependencies:
    "@types/graphql" "0.12.6"
    apollo-utilities "^1.0.0"
    zen-observable-ts "^0.8.9"
```

**package.json:**

```json
  "dependencies": {
    "apollo-cache-inmemory": "1.3.0-verify.4",
    "apollo-cache-persist": "0.1.1",
    "apollo-client": "2.4.2",
    "apollo-link": "1.2.3",
    "apollo-link-context": "1.0.9",
    "apollo-link-error": "1.1.1",
    "apollo-link-http": "1.5.5",
    "apollo-link-persisted-queries": "^0.2.1",
    "apollo-link-state": "0.4.2",
    "apollo-link-ws": "1.0.9",
    "apollo-utilities": "1.0.21",
    "lodash": "4.17.11",
    "react": "16.5.2",
    "react-apollo": "2.1.11",
    "react-dom": "16.5.2",
    "react-router-dom": "4.3.1",
    "react-scripts": "1.1.5",
    "subscriptions-transport-ws": "0.9.15"
  },
  "devDependencies": {
    "graphql": "14.0.2",
    "graphql-tag": "2.9.2",
  },
```

What it can be?

**Comment by JoviDeCroock** [comment]
It seems apollo-link-state is being deprecated, now it's merged with apollo-client (2.5.0). Types graphql is needed to derive types for the AST

**Comment by JoviDeCroock** [comment]
Closing this since it does not involve any of the apollo-link repositories.
